### ISSUE TITLE
`no-unit-unknown` should ignore less variable

### ISSUE THREAD

**Issue by laysent** [stylelint/stylelint#4162]
> Clearly describe the bug

`@56789a` is considered as a number with unknown unit in less, instead of a valid variable name in less.

> Which rule, if any, is the bug related to?

`no-unit-unknown`

> What CSS is needed to reproduce the bug?

```less
@56789a: #56789a; // when you can't find a proper word for this color
a {
  color: @56789a;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-unit-unknown": "always"
  }
}
```

> Which version of stylelint are you using?

10.1.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to LESS variable usage

> What did you expect to happen?

No warning should be reported.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
test.css
2:4    Unexpected unknown unit "a"   unit-no-unknown
```
